MonkeyWrench.Web.WebService/Scheduler/Scheduler.cs

```diff
@@ -400,11 +395,43 @@ private static void AddWork (DB db, List<DBHost> hosts, List<DBLane> lanes, List
 								}
 							}
 
-							sql.AppendFormat ("UPDATE RevisionWork SET state = {0} WHERE id = {1} AND state = 10;", (int) (has_dependencies ? DBState.DependencyNotFulfilled : DBState.NotDone), revisionwork.id);
+							sql.AppendFormat ("UPDATE RevisionWork SET state = 11 WHERE id = {0} AND state = 10;", revisionwork.id);
+							if (!editedHostLanes.Contains (hostlane)) editedHostLanes.Add (hostlane);
+						}
+					}
+				}
 
+				foreach (var editedHostLane in editedHostLanes) {
+					int mostRecent = -1;
+					int mostRecentState = -1;
+					using (IDbCommand cmd = db.CreateCommand ()) {
+						cmd.CommandText = @"
+SELECT RevisionWork.id, RevisionWork.state
+FROM RevisionWork
+INNER JOIN Revision ON Revision.id = RevisionWork.revision_id
+WHERE
+     RevisionWork.host_id = @host_id
+ AND RevisionWork.lane_id = @lane_id
+ORDER BY Revision.Date DESC
+LIMIT 1
+";
+						DB.CreateParameter (cmd, "host_id", editedHostLane.host_id);

```

Comment:
I don't think that's the right query. It gives us a listing of all revision works, since we're grouping by the id (primary key). Furthermore, I think that getting it to work will make it look pretty nasty because of https://stackoverflow.com/questions/7745609/sql-select-only-rows-with-max-value-on-a-column.

This is what I got:
```sql
WITH joined AS (
    SELECT RevisionWork.lane_id, RevisionWork.host_id, RevisionWork.id, Revision.date
    FROM RevisionWork
    INNER JOIN Revision ON RevisionWork.revision_id = Revision.id
),

max_date AS (
    SELECT joined.lane_id, joined.host_id, MAX(joined.date) AS max_date
    FROM joined
    WHERE lane_id = 2716
    GROUP BY joined.lane_id, joined.host_id
)

SELECT * 
FROM joined
INNER JOIN max_date ON joined.date = max_date.max_date AND joined.lane_id = max_date.lane_id AND joined.host_id = max_date.host_id
```